YoCode/FileChangeChecker.cs

```diff
@@ -18,27 +18,36 @@ public static bool ProjectIsModified(IDirectory dir)
                 var originalFileStreams = dir.ReturnOriginalPathFileStream().OrderBy((c => c.path));
                 var modifiedFileStreams = dir.ReturnModifiedPathFileStream().OrderBy((c => c.path));
 
-                var streamsList = originalFileStreams.Zip(modifiedFileStreams, (o, m) => (original: o, modified: m ));
-
-                foreach (var (original, modified) in streamsList)
+                try
                 {
-                    if (FileIsModified(original.content, modified.content))
+                    var streamsList = originalFileStreams.Zip(modifiedFileStreams, (o, m) => (original: o, modified: m));
+
+                    foreach (var (original, modified) in streamsList)
                     {
-                        return true;
+                        if (FileIsModified(original.content, modified.content))
+                        {
+                            return true;
+                        }
                     }
+                    return false;
+                }
+                finally
+                {
+                    foreach (var o in originalFileStreams) o.content.Dispose();
+                    foreach (var m in modifiedFileStreams) m.content.Dispose();
```

Comment:
Like I said at your desk, we need a better solution to this but it will involve bigger changes, so we can leave it for now.